lib/Runtime/PlatformAgnostic/Platform/Linux/HiResTimer.cpp

```diff
@@ -0,0 +1,80 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Runtime.h"
+#include <time.h>
+#include <sys/time.h>
+#include "ChakraPlatform.h"
+
+namespace Js
+{
+    double HiResTimer::GetSystemTime()
+    {
+// todo-osx: mach_absolute_time
+#ifndef __APPLE__
```

Comment:
Considering this is Linux specific (directory is platform\linux), we probably don't need to worry about __apple__. We can do that seperately in platformagnostic\osx and refactor any common code at that time